JavaScript/common.js

```diff
@@ -1,6 +1,32 @@
+// Drop down menu functionality
 const nav_button = document.getElementById("nav_button");
 const nav_bar = document.getElementById("nav_bar");
 
 nav_button.addEventListener('click', () => {
-    nav_bar.classList.toggle('hidden');
-});
\ No newline at end of file
+  nav_bar.classList.toggle('hidden');
+});
+
+// Geolocation Api
+const findState = () => {
+
+	const location = document.getElementById("location");

```

Comment:
What happens if we don't find the element on the page?

Reply:
This is a quick fix. It works but I urge to think a bit broader. 
If I don't have the element, do I need to use the geo web API? 
Do I need to attach an event listeners?

Reply:
Ok, if i don't have the element the API call should be avoided it's only going to slow down the page loading process.
I can attach an even listener by transforming the div location into a button location and im going to call the API only when the user clicks the button rather then calling the API when i load the page...
By doing that we respect the user privacy by not tracking them if not needed?

Reply:
That sounds much better doesn't it? More efficient and more privacy and better UX (since the pop-up with geolocation permission only shows when needed, right?)